op-geth.yaml

```diff
@@ -0,0 +1,189 @@
+      - uses: go/build
+        with:
+          packages: ./cmd/devp2p
+          ldflags: |
+            -X github.com/ethereum/go-ethereum/version.gitTag=v${{package.version}}
+            -X github.com/ethereum/go-ethereum/params.gitTag=v${{package.version}}
+          output: devp2p
+    test:
+      pipeline:
+        - runs: |
+            devp2p --version
+            devp2p --help
+        - runs: |
+            devp2p key generate mynode.key
+
+            # Verify that the key was created and contains a valid hex string
+            if [ ! -f "mynode.key" ]; then
+              echo "Error: mynode.key was not created!" >&2
+              exit 1
+            fi
+
+            KEY_CONTENT=$(cat mynode.key)
+            if ! echo "$KEY_CONTENT" | grep -qE '^[a-f0-9]{64}$'; then
+              echo "Error: Invalid key format in mynode.key" >&2
+              exit 1
+            fi
+
+            echo "Successfully generated and validated mynode.key"
+
+  - name: ${{package.name}}-evm
+    pipeline:
+      - uses: go/build
+        with:
+          packages: ./cmd/evm
+          ldflags: |
+            -X github.com/ethereum/go-ethereum/version.gitTag=v${{package.version}}
+            -X github.com/ethereum/go-ethereum/params.gitTag=v${{package.version}}
+          output: evm
+    test:
+      pipeline:
+        - runs: |
+            evm --version
+            evm --help
+            evm run --code 0x 2>/dev/null && echo "Success" || echo "Failed"
+
+  - name: ${{package.name}}-rlpdump
+    pipeline:
+      - uses: go/build
+        with:
+          packages: ./cmd/rlpdump
+          output: rlpdump
+    test:
+      pipeline:
+        - runs: |
+            rlpdump --help
+            echo "c0" | rlpdump | grep -q '"c"' && echo "rlpdump stdin test passed" || { echo "rlpdump stdin test failed"; exit 1; } # Test: Decode hex "c0" using stdin and validate output
+            rlpdump -hex c0 | grep -q '\[\]' && echo "rlpdump hex test passed" || { echo "rlpdump hex test failed"; exit 1; }
+
+test:
+  pipeline:
+    - name: version and help tests for geth
+      runs: |
+        geth version | grep -qi "${{package.version}}-stable"
+        geth --help
+    - name: "Testing geth process"
+      uses: test/daemon-check-output
+      with:
+        start: geth --dev --nodiscover --maxpeers 0 --http --http.addr "127.0.0.1" --http.port "8545" --http.api "eth,net,web3" --allow-insecure-unlock
+        expected_output: |
+          Maximum peer count
+          HTTP server started
+          Writing custom genesis block
+          Using developer account
+        error_strings: |
+          "panic"
+          "fatal"
+          failed
+          denied
+        post: |
+          sleep 5
+          # Check if the process is running
+          if ps aux | grep -q "[g]eth"; then
+            echo "Geth started successfully"
+          else
+            echo "Failed to start Geth"
+            exit 1
+          fi
+          pkill -f geth
```

Comment:
It looks like this block redundant. This pipeline already checks the process existence and kills on cleanup. We should actually _do_ functional tests here.